client/src/components/NavBar.js

```diff
@@ -80,7 +80,7 @@ function NavBar(props) {
   // navbar for when logged in
   function loggedInNavLinks() {
     // get username from local storage
-    const username = window.localStorage.getItem("username"); 
+    const username = window.localStorage.getItem("username");
```

Comment:
Similar to @alaindavis90 other PR, I don't love this pattern of saving this infomration in localStorage. I think I'm okay with it right now, but here is why: 
1. What if you need more information about the user in the NavBar? Would you continue storing it in localStorage?
2. The token actually has a bunch of information about the user in it. The usually pattern should be like this:
- Token is stored in local or session storage
- In a higher level component, when the user is logged in, you make an API request to get the user. If the token is right, it gets the right user, and then that user details is passed in props to NavBar to give the Navbar information about the user.
- If the user is not logged in, you can respond in that fetch request with the right action, like redirect to sign up / sign in page

Reply:
Okay yeah storing username in localStorage is on me. So right now the navbar is regenerated on each page you visit, taking this into consideration, I see a few ways to do this:
1.  fetch the necessary user data on each page you visit and pass it as props to the navbar
2.  fetch the data in the navbar component
3.  fetch user data once logged in and pass the data as props to each page you visit (and back) and pass that as props to the navbar 

Which one seems more reasonable?

Reply:
so you'd probably do something like the 3rd. Definitely not the 2nd. 

You'd have one higher level component for everything, and it would hold the navbar and then routes for the different pages when logged in
That higher level component would fetch the user once logged in, and would pass it's information as props to each page and the navbar

Reply:
Makes sense! I will make the changes in a future branch.